### PR TITLE
Bump scipy minimal version to 0.14

### PR DESCRIPTION
**Pull request by GaelVaroquaux**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,18 +20,18 @@ matrix:
   include:
     # Ubuntu 14.04 versions (with neurodebian)
     - env: DISTRIB="conda" PYTHON_VERSION="2.7"
-           NUMPY_VERSION="1.8.2" SCIPY_VERSION="0.13.3"
+           NUMPY_VERSION="1.8.2" SCIPY_VERSION="0.14.2"
```
Comment (KamalakerDadi) [inline .travis.yml]: I don't think there is 0.14.2 ?

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,5 @@
-dist: trusty
 sudo: required
+dist: xenial
```
Comment (KamalakerDadi) [inline .travis.yml]: Does this needs to be added above sudo: required ?
https://docs.travis-ci.com/user/reference/trusty/

I had a similar problem while bumping scikit-learn to 0.15. That's why added dist: trusty

**Comment by GaelVaroquaux** [comment]
> Does this needs to be added above sudo: required ?
> https://docs.travis-ci.com/user/reference/trusty/

I think that actually xenial is not supported on travis :(

**Comment by KamalakerDadi** [comment]
>Does this needs to be added above sudo: required ?
 https://docs.travis-ci.com/user/reference/trusty/
>I think that actually xenial is not supported on travis :(

Is it good to move to xenial ?

**Comment by GaelVaroquaux** [comment]
Wow, I uncovered few bugs by changing the versions on the CI. This was probably useful.

**Comment by KamalakerDadi** [comment]
>Wow, I uncovered few bugs by changing the versions on the CI.

Was it with resample or fast_svd ?

**Comment by GaelVaroquaux** [comment]
>     I think that actually xenial is not supported on travis :(

> Is it good to move to xenial ?

It would be. But we cannot.

**Comment by GaelVaroquaux** [comment]
> Was it with resample or fast_svd ?

Both

File: doc/whats_new.rst [hunk 3]
```diff
@@ -4,6 +4,8 @@
 Changelog
 ---------
 
+    - The minimum required version for scipy is now 0.14.1
```
Comment (KamalakerDadi) [inline doc/whats_new.rst]: Probably this should be 0.14 now.

**Comment by GaelVaroquaux** [comment]
OK, merging.

**Pull request merged**